modules/javafx.controls/src/main/java/javafx/scene/control/skin/TextFieldSkin.java

```diff
@@ -391,6 +386,8 @@ public TextFieldSkin(final TextField control) {
 
     /** {@inheritDoc} */
     @Override public void dispose() {
+        if (getSkinnable() == null) return;
```

Comment:
Just curious: Can the skinnable be null here? And is it fine then to never call **super.dispose()** ?

Reply:
dispose has no precondition - it can be called multiple times, so has to guard itself against having cleaned already. And super is called the first time.